src/3box.js

```diff
@@ -523,6 +532,32 @@ class Box {
     utils.fetchJson(this._serverUrl + '/link', linkData).catch(console.error)
   }
 
+  async _writeAddressLink(data) {
+    const link = {
+      type: 'address-link',
+      data
+    }
+    await this._rootStore.add(link)
+  }
+
+  async _readAddressLinks() {
+    const entries = await this._rootStore.iterator({ limit: -1 }).collect()
+    const linkEntries = entries.filter(e => e.payload.value.type === 'address-link')
+    return linkEntries.map(async (entry) => {
+      // TODO handle missing ipfs obj??, timeouts?
```

Comment:
We should try to make sure that this doesn't happen. Seems quite difficult to deal with.